node/src/stream_handler_pool.rs

```diff
@@ -367,218 +370,352 @@ impl StreamHandlerPool {
+            Err(e) => {
                 debug!(
                     self.logger,
-                    "No existing stream keyed by {}: creating one to {}", sw_key, peer_addr
+                    "Removing channel to disabled StreamWriter {} to {}: {}", sw_key, peer_addr, e
                 );
+                // TODO GH-667 It looks like what we should do here is inform our caller somehow
+                // that we can no longer communicate with the Node specified in the route, and
+                // signal that somebody (the Dispatcher?) should remove this neighbor, make another
+                // route, and try again.
+                return Ok(true);
+            }
+            Ok(_) => {
+                debug!(self.logger, "Queued {} bytes for transmission", packet_len);
+            }
+        };
+        if msg.context.last_data {
+            debug!(
+                self.logger,
+                "Removing channel to StreamWriter {} to {} in response to server-drop report",
+                sw_key,
+                peer_addr
+            );
+            return Ok(true);
+        }
+        Ok(false)
+    }
 
-                let subs = self
-                    .self_subs_opt
-                    .clone()
-                    .expect("StreamHandlerPool Unbound");
-                let add_stream_sub = subs.add_sub;
-                let node_query_response_sub = subs.node_query_response;
-                let connection_progress_sub_ok = self
-                    .connection_progress_sub_opt
-                    .clone()
-                    .expect("Neighborhood Unbound");
-                let connection_progress_sub_err = self
-                    .connection_progress_sub_opt
-                    .clone()
-                    .expect("Neighborhood Unbound");
-                let remove_stream_sub = subs.remove_sub;
-                let remove_neighbor_sub = self
-                    .remove_neighbor_sub_opt
-                    .clone()
-                    .expect("Neighborhood Unbound");
+    fn delay_packet_for_opening_stream(
+        &self,
+        msg: DispatcherNodeQueryResponse,
+        peer_addr: SocketAddr,
+        sw_key: StreamWriterKey,
+    ) {
+        debug!(
+            self.logger,
+            "Found in-the-process-of-being-opened stream to {} keyed by {}: preparing to use",
+            peer_addr,
+            sw_key
+        );
+        // a connection is already in progress. resubmit this message, to give the connection time to complete
+        info!(
+            self.logger,
+            "connection for {} in progress, resubmitting {} bytes",
+            peer_addr,
+            msg.context.data.len()
+        );
+        let recipient = self
+            .self_subs_opt
+            .as_ref()
+            .expect("StreamHandlerPool is unbound.")
+            .node_query_response
+            .clone();
+        // TODO FIXME revisit once SC-358/GH-96 is done (idea: use notify_later() to delay messages)
```

Comment:
Maybe explore `notify_later()` here; if it's easy, do it in this card, and if it's hard, make another card for it.

Reply:
Apparently, it'll require us to invent a new message, which can schedule a `notify_later()`. It would be better if this new message stays generic and can be used in other parts of the codebase.

I believe, it deserves a new card. So, I created this card - https://github.com/MASQ-Project/Node-issues/issues/678